src/Npgsql/NpgsqlConnection.cs

```diff
@@ -501,16 +501,17 @@ public new NpgsqlTransaction BeginTransaction(IsolationLevel level)
         /// <summary>
         /// Enlist transation.
         /// </summary>
-        public override void EnlistTransaction(Transaction transaction)
+        public override void EnlistTransaction([CanBeNull]Transaction transaction)
         {
-            if (EnlistedTransaction != null)
+            var enlistedTransaction = EnlistedTransaction;
+            if (enlistedTransaction != null)
             {
-                if (EnlistedTransaction.Equals(transaction))
+                if (enlistedTransaction.Equals(transaction))
                     return;
                 try
                 {
-                    if (EnlistedTransaction.TransactionInformation.Status == System.Transactions.TransactionStatus.Active)
-                        throw new InvalidOperationException($"Already enlisted to transaction (localid={EnlistedTransaction.TransactionInformation.LocalIdentifier})");
+                    if (enlistedTransaction.TransactionInformation.Status == System.Transactions.TransactionStatus.Active)
+                        throw new InvalidOperationException($"Already enlisted to transaction (localid={enlistedTransaction.TransactionInformation.LocalIdentifier})");
```

Comment:
> New bug discovered: after scope disposal in distributed case, the transaction status may still be Active!

I actually thought that could be the case simply because of the asynchronous nature of the DTC implementation...